builder/chroots.go

```diff
@@ -0,0 +1,637 @@
+package builder
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/go-ini/ini"
+	"github.com/pkg/errors"
+)
+
+// TODO: Move this to the more general configuration handling.
+type buildChrootsConfig struct {
+	// [Server] section.
+	HasServerSection bool
+	DebugInfoBanned  string
+	DebugInfoLib     string
+	DebugInfoSrc     string
+
+	// [swupd] section.
+	UpdateBundle string
+	ContentURL   string
+	VersionURL   string
+	// Format is already in b.Format.
+}
+
+// TODO: Move this to the more general configuration handling.
+func readBuildChrootsConfig(path string) (*buildChrootsConfig, error) {
+	iniFile, err := ini.InsensitiveLoad(path)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg := &buildChrootsConfig{}
+
+	// TODO: Validate early the fields we read.
+	server, err := iniFile.GetSection("Server")
+	if err == nil {
+		cfg.HasServerSection = true
+		cfg.DebugInfoBanned = server.Key("debuginfo_banned").Value()
+		cfg.DebugInfoLib = server.Key("debuginfo_lib").Value()
+		cfg.DebugInfoSrc = server.Key("debuginfo_src").Value()
+	}
+
+	swupd, err := iniFile.GetSection("swupd")
+	if err != nil {
+		return nil, fmt.Errorf("error in configuration file %s: %s", path, err)
+	}
+
+	getKey := func(section *ini.Section, name string) (string, error) {
+		key, kerr := section.GetKey(name)
+		if kerr != nil {
+			return "", fmt.Errorf("error in configuration file %s: %s", path, kerr)
+		}
+		return key.Value(), nil
+	}
+
+	cfg.UpdateBundle, err = getKey(swupd, "BUNDLE")
+	if err != nil {
+		return nil, err
+	}
+	cfg.ContentURL, err = getKey(swupd, "CONTENTURL")
+	if err != nil {
+		return nil, err
+	}
+	cfg.VersionURL, err = getKey(swupd, "VERSIONURL")
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func (b *Builder) buildBundleChroots(set *bundleSet) error {
+	var err error
+	stateDir := b.Statedir
+	chrootDir := filepath.Join(stateDir, "image")
```

Comment:
It'd be nice to have b.ChrootDir and whatever else is needed setup from the beginning wouldn't it? Out of scope for this PR but reading through made me think of it.

Reply:
OK.